### PR TITLE
Add Insta bot

### LINKED ISSUE #450: Insta Bot
# Aim
- What is the objective of the Script
Insta Bot
# Details

- What the features will your script have
The Script will do the below tasks
To automatically follow a group of people.
To automatically unfollow a group of people.

### PR DESCRIPTION
**Pull request by muthuannamalai12**
# Description
I have added an Insta Bot which does the tasks mentioned below:
To automatically follow a group of people.
To automatically unfollow a group of people.
Please include a summary of the change and which issue is fixed. List any dependencies that are required for this change.

## Fixes #450 

Replace `issue_no` with the issue number which is fixed in this PR

## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?

- [x] Yes
- [ ] No

## Type of change

_Please delete options that are not relevant._

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [x] Documentation Update

## Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [x] My changes generate no new warnings
- [x] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works.

### CODE REVIEW

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
Create a JSON file for storing the username, password, and dictionary of people for follow/unfollow. It will reduce the number of files.

**Comment by muthuannamalai12** [comment]
I am new to creating a JSON File. I have created one below If it is wrong kindly tell me so that I can correct it and learn how to do it correctly. I have named the file as insta.json @kaustubhgupta 

{
    "Username": "sample-username"
},
{
    "Password": "sample-password"
},
[
  {
    "Username": "chitra_grapher",
    "Name": "sngs"
  },
  {
    "Username": "zendaya",
    "Name": "adfh"
  }
]

**Comment by kaustubhgupta** [comment]
> I am new to creating a JSON File. I have created one below If it is wrong kindly tell me so that I can correct it and learn how to do it correctly. I have named the file as insta.json @kaustubhgupta
> 
> {
> "Username": "sample-username"
> },
> {
> "Password": "sample-password"
> },
> [
> {
> "Username": "chitra_grapher",
> "Name": "sngs"
> },
> {
> "Username": "zendaya",
> "Name": "adfh"
> }
> ]

I think this type of JSON should work for your case:
```json
{
    "user_data": {
        "user_name": "user_name_main",
        "user_password": "user_password_main"
    },

    "target_data": {
        "user_1": {
            "user_name_1": "user_name_1",
            "user_actual_name_1": "user_actual_name_1"
        },

        "user_2": {
            "user_name_2": "user_name_2",
            "user_actual_name_2": "user_actual_name_2"
        }
    } 
}
```

**Comment by muthuannamalai12** [comment]
I have updated it @kaustubhgupta

File: insta-bot/insta.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (kaustubhgupta) [inline insta-bot/insta.json:1]: This was a template JSON! You need to put some fake testing data here

File: insta-bot/instabot.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
+from selenium import webdriver
+from selenium.webdriver.chrome.options import Options
+from selenium.common.exceptions import NoSuchElementException
+from pandas import read_csv
+from time import sleep
+
+
+def getCredentials():
```
Comment (kaustubhgupta) [inline insta-bot/instabot.py:8]: As you have deleted the non required files and adapted the JSON format, don't you think all these functions need to be updated?

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

File: insta-bot/instabot.py (hunk 2 above)
Comment (muthuannamalai12) [reply insta-bot/instabot.py:8]: Sorry for it. It was my mistake I should have updated it

**Comment by muthuannamalai12** [comment]
I have updated @kaustubhgupta. Kindly review again

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
There are a lot of mistakes in this script: 
- The readme file is not updated with the latest changes you made. Now, the user data is fetched via JSON file and there is no mention of this. Instead, there are still instructions for the username.txt file which is deleted after the suggestion!
- There is no mention of the chrome driver path and how to set it. Alternatively, you could have given the exact path of the chrome driver like this: 
```python
 driver = webdriver.Chrome('.\insta-bot\chromedriver.exe', chrome_options=option) 
```
- In the instabot.py file, you have not used the dictionary advantage of JSON files and instead made wrong implementations to retrieve data from JSON files. I suggested JSON as it is easier to get data from it as we can access it as a dictionary in Python. Here is an example of that:
```python
import json

with open('./insta-bot/insta.json') as f:
  data = json.load(f)

print(data['user_data']['user_name'])
```
This will return: sample-username

- Even if I fix the JSON error locally, the bot is still not working:
image

**Comment by kaustubhgupta** [comment]
@muthuannamalai12 you haven't responded about this in 2 days. I can give you the last 6 Hrs to fix these issues after that this PR will be closed.

**Comment by muthuannamalai12** [comment]
> @muthuannamalai12 you haven't responded about this in 2 days. I can give you the last 6 Hrs to fix these issues after that this PR will be closed.

I have been trying to correct it I will do my level best If I cannot make it I will close the Pull Request

**Comment by kaustubhgupta** [comment]
> > @muthuannamalai12 you haven't responded about this in 2 days. I can give you the last 6 Hrs to fix these issues after that this PR will be closed.
> 
> I have been trying to correct it I will do my level best If I cannot make it I will close the Pull Request

This script is not even running for one case, didn't you tested this before pushing the code?

**Pull request closed without merging**